docs/en/docs/encoders.md

```diff
@@ -0,0 +1,202 @@
+```
+
+**Example**
+
+Create and register an encoder that handles `msgspec.Struct` types.
+
+```python
+{!> ../../../docs_src/encoders/msgspec.py !}
+```
+
+Simple right? Because now the `MsgSpecEncoder` is registered, you can simply do this in your handlers
+and return **directly** the `msgspec.Struct` object type.
+
+```python
+from msgspec import Struct
+
+from lilya.routing import Path
+
+
+class User(Struct):
+    name: str
+    email: str
+
+
+def msgspec_struct():
+    return User(name="lilya", url="example@lilya.dev")
+```
+
+### Design specific custom encoders
+
+**Lilya being 100% pure python and not tight to any particular validation library** allows you to
+design custom encoders that are later used by Lilya responses.
+
+Ok, this sounds a bit confusing right? I bet it does so let us go slowly.
+
+Imagine you want to use a particular validation library such as [Pydantic](https://pydantic.dev/),
+[msgspec](https://jcristharif.com/msgspec/) or even [attrs](https://www.attrs.org/en/stable/) or something
+else at your choice.
+
+You want to make sure that if you return a pydantic model or a msgspec Struct or even a `define` attr class.
+
+Let us see how it would look like for all of them.
+
+**For pydantic**
+
+Nothing required anymore. Works out of the box thanks to the ModelDumpEncoder. But we can do an instance check instead:
+
+```python
+{!> ../../../docs_src/encoders/pydantic.py !}
+```
+
+
+**For msgspec Struct**
+
+```python
+{!> ../../../docs_src/encoders/msgspec.py !}
+```
+
+**For attrs**
+
+```python
+{!> ../../../docs_src/encoders/attrs.py !}
+```
+
+Easy and poweful, right? Yes.
+
+Do you understand what does this mean? Means you can design **any encoder** at your choice using
+also any library of your choice as well.
+
+The flexibility of Lilya allows you to be free and for Lilya not to be tight to any particular
+library.
+
+!!! Tip
+    You can replace other Encoders by providing a name attribute.
+    By default all encoders use their class-name as name.
+
+#### Custom encoders and responses
+
+After the [custom encoders in the examples](#build-a-custom-encoder) are created, this allows to
+do something like this directly.
+
+```python
+{!> ../../../docs_src/encoders/responses.py !}
+```
+
+#### Custom encoders and the `make_response`
+
+Well, here its where the `make_response` helps you. The `make_response` will generate a `JSONResponse`
+by default and when you return a custom encoder type, there are some limitations to it.
+
+For example, what if you want to return with a different `status_code`? Or even attach a [task](./tasks.md)
+to it?
+
+The custom encoder **does not handle** that for you but the `make_response` does!
+
+Let us see how it would look like now using the `make_response`.
+
+```python
+{!> ../../../docs_src/encoders/make_response.py !}
+```
+
+
+## Use with custom json encoder
+
+Sometimes you want speed and replace the default json encoder for e.g. orjson.
```

Comment:
Sometimes you might want a more performant json parser that is not the default python `json` built-in, for example, the `orjson`,